core/primitives/src/shard_layout.rs

```diff
@@ -135,6 +334,46 @@ impl ShardLayout {
         })
     }
 
+    /// Return a V2 Shardlayout
+    pub fn v2(
+        account_range_shard_mapping: AccountRangeShardMap,
+        shards_split_map: Option<ShardsSplitMapV2>,
+        version: ShardVersion,
+    ) -> Self {
+        validate_account_range_shard_mapping(&account_range_shard_mapping).unwrap();
+
+        let Some(shards_split_map) = shards_split_map else {
+            return Self::V2(ShardLayoutV2 {
+                account_range_shard_mapping,
+                shards_split_map: None,
+                shards_parent_map: None,
+                version,
+            });
+        };
+
+        let mut shards_parent_map = ShardsParentMapV2::new();
+        for (&parent_shard_id, shard_ids) in shards_split_map.iter() {
```

Comment:
I'm wondering if at some point we would need to implement a check such as: at most one shard is being split in a given ShardLayout. Maybe another day :)

Reply:
I'm wondering if we can perform a few splits at the same time :)

Reply:
Maybe another day :)